Documentation/contributing/development/dev_setup.rst

```diff
@@ -331,6 +350,68 @@ Updating k8s is a special case, for that one needs to do:
     $ make generate-k8s-api
     $ git add go.mod go.sum vendor/
 
+Add/update a new Kubernetes version
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Lets assume we want to add a new Kubernetes version ``v1.19.0``:
+
+#. Follow the above instructions to update the Kubernetes libraries.
+
+#. Follow the next instructions depending if it is a minor update or a patch
+   update.
+
+Minor version
+^^^^^^^^^^^^^
+
+#. Check if it is possible to remove the last supported Kubernetes version from
+   :ref:`k8scompatibility`, :ref:`k8s_requirements`, :ref:`test_matrix`,
+   :ref:`running_k8s_tests`, and add the new Kubernetes version to that list.
+
+#. If the minimal supported version changed, leave a note in the upgrade guide
+   stating the minimal supported Kubernetes version.
+
+#. If the minimal supported version changed, search over the code, more likely
+   under ``pkg/k8s``, if there is code that can be removed which specifically
+   exists for the compatibility of the previous Kubernetes minimal version
+   supported.
+
+#. If the minimal supported version changed, update the field
+   ``MinimalVersionConstraint`` in ``pkg/k8s/version/version.go``
+
+#. Sync all "``slim``" types by following the instructions in
+   ``pkg/k8s/slim/README.md``.
```

Comment:
not related to this PR. Just curious if we can wrap all curl commands in this README.md by  with `go generate` then chuck `go generate` step in Makefile, this will help to automate this step as well.

Reply:
I think this is a good idea. Maybe one for a followup.